code/procs/announce.dm

```diff
@@ -145,4 +145,35 @@ datum/announcement/proc/NewsCast(message as text, message_title as text)
 		return "Service"
 	if(job.department_flag & MIN)
 		return "Mining"
-	return "Common"
\ No newline at end of file
+	return "Common"
+
+/datum/announcement/priority/command/special/Announce(message as text, new_title = "", new_sound = null, do_newscast = newscast, msg_sanitized = 0, zlevels = GLOB.using_map.contact_levels)
+	if(!message)
+		return
+
+	if (contains_links(message))
+		to_chat(usr, SPAN_DANGER("Links are not allowed in in-character messages"))
+		return FALSE
+	var/message_title = new_title ? new_title : title
+	var/message_sound = new_sound ? new_sound : sound
+
+	if(!msg_sanitized)
+		message = sanitize(message, extra = 0, allow_links = FALSE)	//No links in IC communication
+	message_title = sanitizeSafe(message_title)
+
+	var/msg = FormMessage(message, message_title)
+	for(var/mob/M in GLOB.player_list)
+		if(M.is_necromorph() || issignal(M)) continue
```

Comment:
is_necromorph will return true for signals as well, so issignal is not needed here